Web Inspector: Add Versioned Protocol for iOS 18.4 and macOS 15.4

the-chenergy opened the pull request:
#### edc86d5d59affadebd053ac2d909f7ecb6f5a815
<pre>
Web Inspector: Add Versioned Protocol for iOS 18.4 and macOS 15.4
<a href="https://rdar.apple.com/131131024">rdar://131131024</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=288212">https://bugs.webkit.org/show_bug.cgi?id=288212</a>

Reviewed by BJ Burg.

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.4/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/Versions/Inspector-iOS-18.4.json: Added.
* Source/WebInspectorUI/Versions/Inspector-macOS-15.4.json: Added.
   - Add versioned protocol.

* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype._supportsAllAnimationFramesBreakpoint):
* Source/WebInspectorUI/UserInterface/Views/CookieStorageContentView.js:
(WI.CookieStorageContentView.prototype.async _willDismissCookiePopover):
   - Update compatibility comments.

Canonical link: <a href="https://commits.webkit.org/291076@main">https://commits.webkit.org/291076@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2237e2da1ff68971fd6b3304e02d95004347316e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91958 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11489 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1034 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/96892 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/42562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/11835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19975 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/96892 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/42562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94959 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/11835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/83291 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/96892 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/11835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/887 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/41777 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/11835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98922 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/19079 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/14102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/79587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/19331 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79138 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78814 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19510 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/12117 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/19060 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/24269 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/18757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/22216 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/20508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js:
```diff
@@ -0,0 +1,589 @@
+InspectorBackend.registerAnimationDispatcher = InspectorBackend.registerDispatcher.bind(InspectorBackend, "Animation");
+InspectorBackend.activateDomain("Animation", ["page", "web-page"]);
+
+// ApplicationCache
```
all mentions of `ApplicationCache` and `appcache` and etc. everywhere in this PR should be removed since the frontend no longer supports that as of [289820@main](https://commits.webkit.org/289820@main)

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js:
```diff
@@ -0,0 +1,589 @@
+InspectorBackend.registerEnum("DOMDebugger.DOMBreakpointType", {SubtreeModified: "subtree-modified", AttributeModified: "attribute-modified", NodeRemoved: "node-removed"});
+InspectorBackend.registerEnum("DOMDebugger.EventBreakpointType", {AnimationFrame: "animation-frame", Interval: "interval", Listener: "listener", Timeout: "timeout"});
+InspectorBackend.registerCommand("DOMDebugger.setDOMBreakpoint", null, [{"name": "nodeId", "type": "number"}, {"name": "type", "type": "string"}, {"name": "options", "type": "object", "optional": true}], []);
+InspectorBackend.registerCommand("DOMDebugger.removeDOMBreakpoint", null, [{"name": "nodeId", "type": "number"}, {"name": "type", "type": "string"}], []);
```
these should both be limited to `["page"]` in both the `.js` and `.json` as of [290245@main](https://commits.webkit.org/290245@main)

rcaliman-apple replied:
~That commit didn't make it into iOS 18.4 / macOS 15.4. It's correct that it's not reflected in the snapshot.~

Oh, now I notice that patch updated all the snapshots (so we'd want to update this as well). But what was the reason for manually changing old protocol snapshots?

rcaliman-apple replied:
From https://github.com/WebKit/WebKit/pull/39675
>Drive-by: mark `DOMDebugger.setDOMBreakpoint` and `DOMDebugger.removeDOMBreakpoint` as being only supported by `"page"` targets since that's how the code behaves (i.e. it'll throw an error if dispatched to a `Worker`).

Does this mean that setting a breakpoint in a Worker when inspecting an older WebKit build would error out?

rcaliman-apple replied:
Resurrecting the original comment:

https://commits.webkit.org/290245@main didn't make it into iOS 18.4 / macOS 15.4 so its changes shouldn't be included in the snapshot.

dcrousso replied:
yeah in the backend it was an error, so this just codifies that reality and allows the frontend to use `target.hasCommand("...")` instead of a more manual check based on the `target.type`

rcaliman-apple replied:
Acknowledged. That patch (https://commits.webkit.org/290245@main) didn't land in time for iOS 18.4. We'll make a note to merge back the protocol snapshot rewrites when making the snapshot for the release where it's included.

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js:
```diff
@@ -0,0 +1,589 @@
+InspectorBackend.registerDOMStorageDispatcher = InspectorBackend.registerDispatcher.bind(InspectorBackend, "DOMStorage");
+InspectorBackend.activateDomain("DOMStorage", ["itml", "page", "web-page"]);
+
+// Database
```
all mentions of `Database` everywhere in this PR should be removed since the frontend no longer supports that as of [289854@main](https://commits.webkit.org/289854@main)

dcrousso reviewed (changes requested):
this is mostly good, but needs a few things fixed before it's ready to merge

please find and replace all `iOS 18.X` and `macOS 15.X` with `iOS 18.4` and `macOS 15.4` respectively.  there should only be in a few spots, specifically those dealing with setting partitioned cookies and creating `requestAnimationFrame` breakpoints in `Worker`

also, there's some manual adjustments needed to these generated files due to some recent changes

the-chenergy committed:
Web Inspector: Add Versioned Protocol for iOS 18.4 and macOS 15.4
rdar://131131024
https://bugs.webkit.org/show_bug.cgi?id=288212

Reviewed by BJ Burg.

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.4/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/Versions/Inspector-iOS-18.4.json: Added.
* Source/WebInspectorUI/Versions/Inspector-macOS-15.4.json: Added.
   - Add versioned protocol.

* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype._supportsAllAnimationFramesBreakpoint):
* Source/WebInspectorUI/UserInterface/Views/CookieStorageContentView.js:
(WI.CookieStorageContentView.prototype.async _willDismissCookiePopover):
   - Update compatibility comments.

Canonical link: https://commits.webkit.org/291076@main
